json_path/src/json_path.rs

```diff
@@ -860,7 +925,7 @@ impl<'i, UPTG: UserPathTrackerGenerator> PathCalculator<'i, UPTG> {
                 Some(term) => {
                     let mut calc_data = PathCalculatorData {
                         results: Vec::new(),
-                        root: json,
+                        root: json.clone(),
```

Comment:
Can these be handled without potentially cloning the entire document? Consider adding a TODO here and in all the subsequent `.clone()`s in this file. There is quite a lot of cloning occurring and I'm not certain that all of it is desirable or necessary.